src/qtism/data/storage/xml/XmlDocument.php

```diff
@@ -677,4 +684,34 @@ protected function inferVersion(): string
     {
         return QtiVersion::infer($this->getDomDocument());
     }
+
+    /**
+     * Validate the document component after unmarshalling.
+     * @throws BranchRuleTargetException
+     */
+    private function validateDocComponent(QtiComponent $docComponent): void
+    {
+        $branchRules = $docComponent->getComponentsByClassName('branchRule', true);
+        if ($branchRules->count() === 0) {
+            return;
+        }
+
+        $errors = [];
+        foreach ($branchRules as $branchRule) {
+            $target = $branchRule->getTarget();
+            if (empty($target)) {
+                $errors[] = 'BranchRule is missing a target attribute';
```

Comment:
There could be multiple branch rules but you don't include info which one is failing. Add some name or key in the error message.